src/app/util/ember-compatibility-functions.cpp

```diff
@@ -516,6 +516,43 @@ Protocols::InteractionModel::Status UnsupportedAttributeStatus(const ConcreteAtt
 
 } // anonymous namespace
 
+template <typename T>
```

Comment:
This templat'ization is likely going to result in a lot of bloat if the compiler doesn't get this right for each specialization. Can we re-factor this to have the bulk of the common bits of logic in a separate function, and to call that from here?